File: Dynamic_Programming/baekjoon_10844.py
def stair(n):
    
    num = int(1e9) 
    dp = [[0]*10 for _ in range(101)]
    
    # dp[i][j] => 길이가 i이고 i번째 자리수의 숫자가 j인 경우의 수
    
    for i in range(1,10):
        dp[1][i] = 1
        
    for i in range(2, 101):
        for j in range(10):
            if j == 0:
                dp[i][j] = dp[i-1][1]
            
            elif j == 9:
                dp[i][j]= dp[i-1][8]
                
            else: 
                dp[i][j] = dp[i-1][j-1] + dp[i-1][j+1]  
    
    ans = sum(dp[n])  
    
    return ans%num

if __name__ == "__main__":
    
    n = int(input())
    if n<1 or n>100:
        raise Exception("0<=n<=100")
             
    print(stair(n))

File: Dynamic_Programming/baekjoon_10844.c
#include<stdio.h>
#define num 1000000000

int main(){

    int n;
    int dp[101][10]= {};
    scanf("%d", &n);

    for(int i=1; i<10; i++){
        dp[1][i] = 1;
    }

    for(int j=2; j<n+1; j++){
        for(int k=0; k<10; k++){
            if(k == 0){
                dp[j][0] = dp[j-1][1];
            }
            else if(k == 9){
                dp[j][9] = dp[j-1][8];
            }
            else{
                dp[j][k] = dp[j-1][k-1] + dp[j-1][k+1];
            }
        }
    }

    int ans = 0;

    for(int q=0; q < 10; q++){
        ans = ans + dp[n][q];
    }

    printf("%d\n", ans%num);
    return 0;
}

